pkg/datapath/loader/netlink.go

```diff
@@ -329,3 +332,58 @@ func SetupBaseDevice(mtu int) (netlink.Link, netlink.Link, error) {
 
 	return linkHost, linkNet, nil
 }
+
+// reloadIPSecOnLinkChanges subscribes to link changes to detect newly added devices
+// and reinitializes IPsec on changes. Only in effect for ENI mode in which we expect
+// new devices at runtime.
+func (l *Loader) reloadIPSecOnLinkChanges() {
+	// settleDuration is the amount of time to wait for further link updates
+	// before proceeding with reinitialization. This avoids back-to-back
+	// reinitialization when multiple link changes are made at once.
+	const settleDuration = 1 * time.Second
+
+	if !option.Config.EnableIPSec || option.Config.IPAM != ipamOption.IPAMENI {
+		return
+	}
+
+	ctx := context.Background()
+	updates := make(chan netlink.LinkUpdate)
+
+	if err := netlink.LinkSubscribe(updates, ctx.Done()); err != nil {
```

Comment:
I guess this is only started once, so the fact that `ctx.Done()` will never be closed is somewhat fine.

But it might be cleaner if we actually cancelled `ctx` if the spawned go routine exits, to avoid leaking subscriptions e.g. if `reloadIPSecOnLinkChanges` is ever called from a unit test.

Reply:
Yeah, the context lives forever as loader is never stopped. I'll add the `defer cancel()` into the spawned goroutine even though the plan is to kill this code as soon as we have cleaner fix (https://github.com/cilium/cilium/pull/24677 and follow-ups to always do runtime detection of devices).